src/image/imageloadobserver.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/image/imageloadobserver
+ */
+
+import DomEventObserver from '@ckeditor/ckeditor5-engine/src/view/observer/domeventobserver';
+import DOMEmitterMixin from '@ckeditor/ckeditor5-utils/src/dom/emittermixin';
+
+/**
+ * Observes all images added to the {@link module:engine/view/document~Document},
+ * fires {@link module:engine/view/document~Document#event:imageLoad}` event and
+ * {@link module:engine/view/view~View#render renders} the View every time when new image
+ * has been loaded.
+ *
+ * **Note:** This event is not fired for images that has been added to the document as `complete` (already loaded).
+ *
+ * @extends module:engine/view/observer/domeventobserver~DomEventObserver
+ */
+export default class ImageLoadObserver extends DomEventObserver {
+	constructor( view ) {
+		super( view );
+
+		/**
+		 * List of img DOM elements that are observed by this observer.
+		 *
+		 * @private
+		 * @type {Set.<HTMLElement>}
+		 */
+		this._observedElements = new Set();
+
+		/**
+		 * DOM emitter used for listening images `load` event.
+		 *
+		 * @private
+		 * @type {module:utils/dom/emittermixin~DomEmitterMixin}
+		 */
+		this._domEmitter = Object.create( DOMEmitterMixin );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	observe( domRoot, name ) {
+		const viewRoot = this.document.getRoot( name );
+
+		viewRoot.on( 'change:children', ( evt, node ) => {
+			this.view.once( 'render', () => {
+				if ( !node.is( 'element' ) || node.is( 'attributeElement' ) ) {
+					return;
+				}
+
+				const domNode = this.view.domConverter.mapViewToDom( node );
+
+				for ( const domElement of domNode.querySelectorAll( 'img' ) ) {
+					if ( !this._observedElements.has( domElement ) ) {
+						this._domEmitter.listenTo( domElement, 'load', ( evt, domEvt ) => this.onDomEvent( domEvt ) );
+						this._observedElements.add( domElement );
+					}
+				}
+
+				// Clean up the list of observed elements from elements that has been removed from the root.
+				for ( const domElement of this._observedElements ) {
+					if ( !domRoot.contains( domElement ) ) {
+						this._domEmitter.stopListening( domElement );
+						this._observedElements.delete( domElement );
+					}
+				}
+			} );
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	onDomEvent( domEvent ) {
+		this.view.render();
+		this.fire( 'imageLoad', domEvent );
```

Comment:
`imageLoaded`?

Reply:
Hm... I forgot that the native image is called `load`. Still... `imageLoad` sounds soooo bad.

Reply:
My first thought was just `load` the same as `Image#load` but `load` is too general, not only an image fires this event. So I added image prefix. But `imageLoaded` sounds ok.